src/nested_pandas/nestedframe/core.py

```diff
@@ -363,25 +485,38 @@ def query(self, expr) -> Self:  # type: ignore[name-defined] # noqa: F821
 
         >>> df.query("mynested.a > 2")
         """
-
-        # Rebuild queries for each specified nested/base layer
-        exprs_to_use = self._split_query(expr)
-
-        # For now (simplicity), limit query to only operating on one layer
-        if len(exprs_to_use.keys()) != 1:
+        if not isinstance(expr, str):
+            msg = f"expr must be a string to be evaluated, {type(expr)} given"
+            raise ValueError(msg)
+        kwargs["level"] = kwargs.pop("level", 0) + 1
+        kwargs["target"] = None
+        # At present, the query expression must be either entirely within the
```

Comment:
Totally fine to keep this behavior, but if you believe this new implementation would support multi-target queries this could be a restriction we'd lift. Maybe opening a separate issue for this would be best

Reply:
I believe this PR makes a significant step toward multi-target queries, but it might be better to make the next step in a separate PR, for simplicity's sake.  Will investigate.

Reply:
Given how long this PR has been in flight, and the amount of complexity in it, I will leave the multi-target queries for a future PR that builds on top of this one.